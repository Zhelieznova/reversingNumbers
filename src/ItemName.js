import React, { Component } from 'react';

export class ItemName extends Component {

    handleChange = (e) => {
        if(isNaN(e.target.value)) alert("Only numbers");
        else this.props.onValueChange(e.target.value);
    };

    render() {
        return (
            <label>
                <br/>
                {this.props.name}
                <input type="text"
                       value={this.props.value}
                       onChange={this.handleChange}/>
            </label>
        );
    }
}
