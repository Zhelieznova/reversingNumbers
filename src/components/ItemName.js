import React, { Component } from 'react';
import PropTypes from 'prop-types';

export default class ItemName extends Component {
  handleChange = (e) => {
    if (isNaN(e.target.value)) {
      alert('sda');
    } else this.props.onValueChange(e.target.value);
  };
  render() {
    return (
      <label>
        <br />
        {this.props.name}
        <input
          type="text"
          value={this.props.value}
          onChange={this.handleChange}
        />
      </label>
    );
  }
}
ItemName.propTypes = {
  name: PropTypes.string.isRequired,
  value: PropTypes.string.isRequired,
  onValueChange: PropTypes.func.isRequired,
};

