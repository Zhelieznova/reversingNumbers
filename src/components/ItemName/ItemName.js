import React, { Component } from 'react';
import PropTypes from 'prop-types';

export default class ItemName extends Component {
  handleChange = (e) => {
    const numberValue = Number(e.target.value);
    if (Number.isNaN(numberValue)) {
      this.props.onError(true);
    } else {
      this.props.onValueChange(e.target.value);
    }
  };
  render() {
    return (
      <label>
        <br />
        {this.props.name}
        <div>Some text</div>
        <input
          type="text"
          value={this.props.value}
          onChange={this.handleChange}
        />
      </label>
    );
  }
}
ItemName.propTypes = {
  name: PropTypes.string,
  value: PropTypes.string,
  onError: PropTypes.func,
  onValueChange: PropTypes.func,
};

