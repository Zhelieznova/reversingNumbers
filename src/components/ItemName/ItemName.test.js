import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json'
import ItemName from './ItemName';

describe('components/ItemName', () => {
  it('renders', () => {
    const tree = shallow(<ItemName name="Normal number: " value="120" />);
    expect(toJSON(tree)).toMatchSnapshot();
  });

  it('calls handleChane with updated target.value', () => {
    const tree = shallow(<ItemName
      name="Normal number: "
      value="120"
      onError={jest.fn()}
    />);
    const instance = tree.instance();
    jest.spyOn(instance, 'handleChange');
    const testData = { target: { value: 'Test' } };
    tree.find('input').prop('onChange')(testData);
    expect(instance.handleChange).toHaveBeenCalledWith(testData);
  });
});
