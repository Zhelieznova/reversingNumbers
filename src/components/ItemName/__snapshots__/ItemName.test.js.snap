// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/ItemName renders 1`] = `
<label>
  <br />
  Normal number: 
  <div>
    Some text
  </div>
  <input
    onChange={[Function]}
    type="text"
    value="120"
  />
</label>
`;
